### PR TITLE
PUBDEV-4593 - Levenstein Distance Normalization Error

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-4593

One "integration" test to prove the original case has been fixed and to complement a standard unit test.

Related master branch PR: https://github.com/h2oai/h2o-3/pull/1911

### CODE REVIEW

File: h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package water.rapids.ast.prims.string.algorithms;
+
+import no.priv.garshol.duke.Comparator;
+
+import static org.apache.commons.lang.math.IEEE754rUtils.min;
+
+/**
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java]: Did the original code have a license in the file?

Ideally we should follow this process: https://opensource.stackexchange.com/a/5161

File: h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+  /**
+   * Computes a case-sensitive Levensthein distance of two strings
+   *
+   * @param a First compared instance of {@link String}
+   * @param b Second compared instance of {@link String}
+   * @return Computed distance between two given strings
+   */
+  private double distance(final String a, final String b) {
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java]: What is the performance of this implementation compared to the original implementation from Duke?

File: h2o-py/tests/testdir_jira/pyunit_pubdev_4593.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+def pubdev_4593():
+    words = ['A', 'AA', 'AAA', 'AAAA', 'AAAAA', 'AAAAAA']
+    compare = ['a', 'aa', 'aaa', 'aaaa', 'aaaaa', 'aaaaaa']
+    words = h2o.H2OFrame(words)
+    compare = h2o.H2OFrame(compare)
+    dist = words.strdistance(compare, 'lv')
+    
+    for row in range(0, dist[0].nrows):
+        assert dist[row, 0] == 0, "Levensthein outcome must be exactly zero"
```
Comment (michalkurka) [inline h2o-py/tests/testdir_jira/pyunit_pubdev_4593.py]: typo in "Levensthein"

File: h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java (hunk 1 above)
Comment (Pscheidl) [reply h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java]: There is Apache 2 license https://github.com/Simmetrics/simmetrics/blob/master/simmetrics-core/src/main/java/org/simmetrics/metrics/Levenshtein.java

File: h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java (hunk 2 above)
Comment (Pscheidl) [reply h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java]: Some hard numbers, as the Big-O is the same. It takes 41 seconds to compare 10 million times a 32-char long:

```java
levenshteinDistanceComparator.compare("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
```

I takes 8 seconds to compare 3-char long string 100 /one hundred/ million times:
```java
      test = levenshteinDistanceComparator.compare("AAAA",
          "aaa");
```

To compare a 1 char long string, it takes half a second for 100 /one hundred/ million times of comparison.

No warmup done.

File: h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java]: Then please include this license in this file

**Pull request merged**